src/agora/agora.h

```diff
@@ -47,6 +45,28 @@ class Agora {
 
   void Start();  /// The main Agora event loop
   void Stop();
+  void GetEqualData(float** ptr, int* size);
+
+  // Flags that allow developer control over Agora internals
+  struct {
+    //     void getEqualData(float** ptr, int* size);Before exiting, save
+    //     LDPC-decoded or demodulated data to a file
+    bool enable_save_decode_data_to_file_ = false;
+
+    // Before exiting, save data sent on downlink to a file
+    bool enable_save_tx_data_to_file_ = false;
+  } flags_;
+
+ private:
+  enum ScheduleProcessingFlags : uint8_t {
+    kNone = 0,
+    kUplinkComplete = 0x1,
+    kDownlinkComplete = 0x2,
+    kProcessingComplete = (kUplinkComplete + kDownlinkComplete)
+  };
+  bool CheckWorkComplete(size_t frame_id);
+  void CheckIncrementScheduleFrame(size_t frame_id,
+                                   ScheduleProcessingFlags completed);
```

Comment:
Could you add comments for the two functions above? The function names are a little confusing to me.

Reply:
Done